contracts/core/fee-calculators/PerformanceFeeCalculator.sol

```diff
@@ -0,0 +1,383 @@
+            feeState[msg.sender].lastProfitFeeTimestamp = block.timestamp;
+        }
+    }
+
+    /**
+     * Parses passed in fee parameters from bytestring.
+     *
+     * 
+     * | CallData                     | Location                      |
+     * |------------------------------|-------------------------------|
+     * | profitFeeFrequency           | 32                            |
+     * | highWatermarkResetFrequency  | 64                            |
+     * | profitFeePercentage          | 96                            |
+     * | streamingFeePercentage       | 128                           |
+     *
+     * @param  _callData            Byte string containing fee parameter data
+     * @return feeParameters        Fee parameters
+     */
+    function parsePerformanceFeeCallData(
+        bytes memory _callData
+    )
+        private
+        pure
+        returns (InitFeeParameters memory)
+    {
+        InitFeeParameters memory parameters;
+
+        assembly {
+            mstore(parameters,           mload(add(_callData, 32)))   // profitFeeFrequency
+            mstore(add(parameters, 32),  mload(add(_callData, 64)))   // highWatermarkResetFrequency
+            mstore(add(parameters, 64),  mload(add(_callData, 96)))   // profitFeePercentage
+            mstore(add(parameters, 96),  mload(add(_callData, 128)))  // streamingFeePercentage
+        }
+
+        return parameters;
+    }
+
+    /*
+     * Validates fee parameters. Ensures that both fees are below the max fee percentages and that they are
+     * multiples of a basis point. Also makes sure highWatermarkResetFrequency is greater than profitFeeFrequency.
+     */
+    function validateFeeParameters(
+        InitFeeParameters memory parameters
+    )
+        private
+        view
+    {
+        require(
+            parameters.profitFeePercentage <= maximumProfitFeePercentage,
+            "PerformanceFeeCalculator.validateFeeParameters: Profit fee exceeds maximum."
+        );
+
+        require(
+            parameters.streamingFeePercentage <= maximumStreamingFeePercentage,
+            "PerformanceFeeCalculator.validateFeeParameters: Streaming fee exceeds maximum."
+        );
+
+        ScaleValidations.validateMultipleOfBasisPoint(parameters.profitFeePercentage);
+        ScaleValidations.validateMultipleOfBasisPoint(parameters.streamingFeePercentage);
+
+        require(
+            parameters.highWatermarkResetFrequency >= parameters.profitFeeFrequency,
+            "PerformanceFeeCalculator.validateFeeParameters: Fee collection frequency must exceed highWatermark reset."
+        );
+    }
+
+    /**
+     * Calculates streaming fee by multiplying streamingFeePercentage by the elapsed amount of time since the last fee
+     * was collected divided by one year in seconds, since the fee is a yearly fee.
+     *
+     * @return uint256        Streaming fee
+     */
+    function calculateStreamingFee()
```

Comment:
Wondering if we would want the resulting fee to be a multiple of a basis point for any reason.

Reply:
Can't think of one in this instance, think thats more of a product thing that we do than any security related reasons.